docs/adr/0002-search-output-ports-implementation.md

```diff
@@ -0,0 +1,57 @@
+# Postgres full text search (FTS) for searching output ports
+
+## Context and Problem Statement
+
+We are introducing full-text search for output ports using PostgreSQL’s built-in capabilities.
+PostgreSQL FTS is simple to implement, performant enough for our current needs, and provides a good foundation for future improvements.
+
+This ADR documents the configuration decisions made for the initial implementation, covering key choices such as ranking,
+field selection, query handling, weighting, indexing, and future extensions.
+
+## Key Decisions to Make
+
+- Which ranking function to use
+- Which fields to include in the search vector
+- How to transform user queries into lexemes
+- Whether to apply different weights per field
+- When to trigger reindexing
+- How to handle varying document lengths
+
+## Decision drivers
+
+We aim for a balanced trade-off between simplicity and quality.
+The initial version should be a strong baseline implementation that can evolve over time.
+We recognize that even advanced search configurations won’t satisfy all edge cases.
+
+## Decisions
+
+### Ranking function
+Use ts_rank_cd for scoring, as it considers term proximity (cover density).
+Alternatives like ts_rank lack this normalization, and a custom ranking function would be overkill at this stage.
+
+### Fields included
+Index only title and description fields of the output port and technical assets.
+These contain the most relevant data; excluding product and tag info keeps indexing simpler.
+
+### Query transformation
+Use websearch_to_tsquery for flexible multi-word queries at the cost of making query syntaxt more complex for users.
```

Comment:
Maybe just add some examples to the ADR?

Reply:
I can do that: I will add a link to the postgres docs and write the rules down.

The rules are:
- unquoted text: text not inside quote marks will be converted to terms separated by & operators, as if processed by plainto_tsquery.
- "quoted text": text inside quote marks will be converted to terms separated by <-> operators, as if processed by phraseto_tsquery.
- OR: the word “or” will be converted to the | operator.
- -: a dash will be converted to the ! operator.